Shared/Samples/Download preplanned map area/DownloadPreplannedMapAreaView.Model.swift

```diff
@@ -181,9 +181,17 @@ class OfflineMapModel: ObservableObject, Identifiable {
     init(preplannedMapArea: PreplannedMapArea, offlineMapTask: OfflineMapTask, temporaryDirectory: URL) {
         self.preplannedMapArea = preplannedMapArea
         self.offlineMapTask = offlineMapTask
-        self.mmpkDirectory = temporaryDirectory
-            .appendingPathComponent(preplannedMapArea.portalItem.id.rawValue)
-            .appendingPathExtension("mmpk")
+        
+        let mmpkDirectory: URL?
+        if let itemID = preplannedMapArea.portalItem.id {
+            mmpkDirectory = temporaryDirectory
+                .appendingPathComponent(itemID.rawValue)
+                .appendingPathExtension("mmpk")
+        } else {
+            mmpkDirectory = nil
```

Comment:
Does this means if the preplanned map area doesn't have a portal ID, then the map doesn't support downloading mmpk at all?

If so, I think we can use a precondition instead of making the directory optional, to alert user this fact.

Reply:
Another idea would be to make this init failable and use `compactMap` instead of `map` at the call site.

Reply:
> Does this means if the preplanned map area doesn't have a portal ID, then the map doesn't support downloading mmpk at all?

I think it just means the preplanned map area is not hosted on AGOL.



> Another idea would be to make this init failable and use compactMap instead of map at the call site.

I like that idea. Because if there is no item ID then we should just not create the model.